scripts/dumpapp

```diff
@@ -24,28 +22,17 @@ def main():
   # transport.
   device = os.environ.get('ANDROID_SERIAL')
   try:
-    conn = HTTPConnectionOverADB(device, process)
-    query_params = ['argv=' + urllib.parse.quote(arg) for arg in args]
-    url = '/dumpapp?%s' % ('&'.join(query_params))
-    fake_host = { 'Host': 'localhost:5037' }
+    sock = stetho_open(device, process)
 
-    http_method = None
-    body = None
-    if not sys.stdin.isatty():
-      http_method = 'POST'
-      body = sys.stdin.detach().read()
-    else:
-      http_method = 'GET'
-      body = None
-
-    conn.request(http_method, url, body, fake_host)
-
-    reply = conn.getresponse()
+    enter_frame = b'!' + struct.pack('!L', len(args))
```

Comment:
With this new architecture it was actually fairly trivial to support the old dumpapp script unmodified.  I added an extra crude HTTP handler that exclusively buffers responses and prints a warning:
$ ./scripts/dumpapp 

```
*** WARNING: Using legacy HTTP protocol; update dumpapp script! ***
Usage: dumpapp [options] <plugin> [plugin-options]
 -h,--help            Print this help
 -l,--list            List available plugins
 -p,--process <arg>   Specify target process
```

The warning is on stderr, so piping the above output to null causes:

```
$ ./scripts/dumpapp -h > /dev/null
*** WARNING: Using legacy HTTP protocol; update dumpapp script! ***
```